### ISSUE TITLE
dialect option instanceName not working as expected

### ISSUE THREAD

**Issue by dsbert** [sequelize/sequelize#8329]
## What are you doing?

I am trying to connect to a non-default instance of sql server with the following config

```json
{
      "dialect": "mssql",
      "host": "127.0.0.1",
      "database": "devdb",
      "databaseVersion": "10.0.0",
      "dialectOptions":{
        "instanceName": "DEV2008"
      }
}
```

## What do you expect to happen?

Based on this pull request #3098 and this issue #3097, I would expect my configuration to work, and Sequelize should connect to the named instance.

## What is actually happening?

The first connection attempt reports a Timeout error. If I retry, I see the following error.

```
"Failed to connect to 127.0.0.1:undefined in 15000ms"
```

It looks like the instance name is not being used, and the port is undefined. I tried this with both the `mssql` package and the `tedious` package.

__Dialect:__ mssql (tedious)
__Dialect version:__ 2.0.1
__Database version:__ 2008 R2
__Sequelize version:__ 4.8.4
__Tested with master branch:__ No

**Comment by dsbert** [comment]
__UPDATE__

It looks like I didn't have the SQL Browser service running. However, upon starting that, I saw an error indicating the following

```
"Port for DEV2008 not found in <list of SQL instances>
```

This error was caused by not having TCP connections enabled for the database instance. Once I enabled that, the connection worked.

Closing this and leaving it for anyone who runs into the same issue in the future. The error messages are not the most intuitive.
